warden/app/genesis.go

```diff
@@ -43,7 +45,7 @@ func NewErc20GenesisState() *erc20types.GenesisState {
 // NOTE: for the example chain implementation we are also adding a default minter.
 func NewMintGenesisState() *minttypes.GenesisState {
 	mintGenState := minttypes.DefaultGenesisState()
-	mintGenState.Params.MintDenom = ChainDenom
+	mintGenState.Params.MintDenom = config.BaseDenom
```

Comment:
we should avoid importing cmd/wardend from app, as the dependency is actually the inverse (cmd/wardend can depend on app, not viceversa)

plus we already have the coin information in `coinInfo`

```suggestion
	mintGenState.Params.MintDenom = coinInfo.Denom
```

Reply:
This is how it's made in evmd)